### PR TITLE
Add bugfix validate check

### LINKED ISSUE #12445: When new tests are added in PR with code changes, also run them on old ClickHouse version and report results.
If new tests did not fail on old ClickHouse version, it may indicate that they does not cover the feature.
But we cannot use this fact to reject pull request, because we also frequently create PRs that add tests for already working code.

CC @alesapin

### PR DESCRIPTION
**Pull request by vdimir**
Changelog category (leave one):

- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- When new tests are added in PR with code changes, also run them on old ClickHouse version and report results, close https://github.com/ClickHouse/ClickHouse/issues/12445

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by vdimir** [comment]
* Commit [`23bee2c` (#35124)](https://github.com/ClickHouse/ClickHouse/pull/35124/commits/23bee2c8435181462ab654ae96e93d7d575653d6) **contains** tests that fail on latest stable branch, check statuses is **ok** ([Stateless Tests report](https://s3.amazonaws.com/clickhouse-test-reports/35124/23bee2c8435181462ab654ae96e93d7d575653d6/stateless_tests_bugfix_validate_check__address__actions_.html), [Integration Tests report](https://s3.amazonaws.com/clickhouse-test-reports/35124/23bee2c8435181462ab654ae96e93d7d575653d6/integration_tests_bugfix_validate_check__asan__actions_.html))

* Commit [`983e8c8` (#35124)](https://github.com/ClickHouse/ClickHouse/pull/35124/commits/983e8c8bdfeb0bb21f2887647ba335f623dbfe2b) **doesn't** contain tests that fail on latest stable branch, check statuses is **fail** ([Stateless Tests report](https://s3.amazonaws.com/clickhouse-test-reports/35124/983e8c8bdfeb0bb21f2887647ba335f623dbfe2b/stateless_tests_bugfix_validate_check__address__actions_.html), [Integration Tests report](https://s3.amazonaws.com/clickhouse-test-reports/35124/983e8c8bdfeb0bb21f2887647ba335f623dbfe2b/integration_tests_bugfix_validate_check__asan__actions_.html))

**Comment by vdimir** [comment]
Commit [`b8b64b1` (#35124)](https://github.com/ClickHouse/ClickHouse/pull/35124/commits/b8b64b1d15f43406dffe75a9f6763d744d71d2c1) makes checks for stateless and integration tests dependent (we should show failed check only if we don't have any type of tests). Will add dummy tests and check if it works

File: tests/ci/functional_test_check.py [hunk 1]
```diff
@@ -139,10 +144,17 @@ def process_results(result_folder, server_log_path):
     check_name = sys.argv[1]
     kill_timeout = int(sys.argv[2])
 
+    validate_bugix_check = len(sys.argv) >= 4 and sys.argv[3] == "--validate-bugfix"
```
Comment (Felixoid) [inline tests/ci/functional_test_check.py:147]: Maybe, it's time to use `argparse`?

**Review by Felixoid — COMMENTED** [review]
Looks good, but a few points to improve

File: tests/ci/functional_test_check.py (hunk 1 above)
Comment (vdimir) [reply tests/ci/functional_test_check.py]: Done

File: tests/ci/download_previous_release.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+CLICKHOUSE_COMMON_STATIC_PACKET_NAME = "clickhouse-common-static_{version}_amd64.deb"
+CLICKHOUSE_COMMON_STATIC_DBG_PACKET_NAME = "clickhouse-common-static-dbg_{version}_amd64.deb"
+CLICKHOUSE_SERVER_PACKET_NAME = "clickhouse-server_{version}_all.deb"
+CLICKHOUSE_CLIENT_PACKET_NAME = "clickhouse-client_{version}_all.deb"
+
+PACKETS_DIR = "previous_release_package_folder/"
+VERSION_PATTERN = r"((?:\d+\.)?(?:\d+\.)?(?:\d+\.)?\d+-[a-zA-Z]*)"
+
+
+class Version:
```
[the remark this answers is not in the document]
Comment (vdimir) [reply tests/ci/download_previous_release.py:33]: Yes, but this file taken from other PR that is not merged, so maybe for now we can leave it without significant changes, not to cause lots of merge conflicts and modify it later, after merging both PRs.

**Comment by vdimir** [comment]
Now it's two independent checks for Integration and Stateless tests. 
It's better to unify it into one check somehow. We can merge this PR to see how it would work and improve later or try to finish directly in this pr, not sure.

**Comment by vdimir** [comment]
Checks for integration and stateless test are performed in one CI check

**Comment by vdimir** [comment]
Checks for integration and stateless test are performed in one CI check

File: tests/ci/pr_info.py [hunk 3]
```diff
@@ -209,6 +209,7 @@ def fetch_changed_files(self):
         else:
             diff_object = PatchSet(response.text)
             self.changed_files = {f.path for f in diff_object}
+        print("Fetched info about %d changed files", len(self.changed_files))
```
Comment (azat) [inline tests/ci/pr_info.py:212]:
```suggestion
        print("Fetched info about", len(self.changed_files), "changed files")
```

**Pull request merged**